nipype/algorithms/confounds.py

```diff
@@ -451,6 +470,12 @@ def _run_interface(self, runtime):
                      'cosine': 'cos'}[self.inputs.pre_filter]
             ncols = filter_basis.shape[1] if filter_basis.size > 0 else 0
             header = ['{}{:02d}'.format(ftype, i) for i in range(ncols)]
+            if nvols:
+                ss_col = np.zeros((components.shape[0], 1),
+                                  dtype=filter_basis.dtype)
+                ss_col[:nvols] = 1
```

Comment:
I think we should model each volume with a separate regressor. Like artifacts in ArtDetect